### PR TITLE
[#113104847] if the story requester doesn't exist in the tracker_users table, pushes for a branch with that story_id error out and dont get run

### PR DESCRIPTION
**Pull request by kquet**
[Tracker #113104847](https://www.pivotaltracker.com/story/show/113104847)

### CODE REVIEW

**Comment by kquet** [comment]
@suwyn

File: lib/integrity/branch.rb [hunk 1]
```diff
@@ -30,11 +31,13 @@ def set_story
...
       return unless story_id
 
-      if Tracker::Story.get(story_id)
-        self.story = Tracker::Story.get(story_id)
+      if story = Tracker::Story.get(story_id)
+        self.story = story
       else
         self.story = Tracker::Story.create(story_id)
       end
+    rescue => e
```
Comment (suwyn) [inline lib/integrity/branch.rb]: You're rescuing all standard errors and assuming they are errors resulting from tracker.    This [could be done better](http://daniel.fone.net.nz/blog/2013/05/28/why-you-should-never-rescue-exception-in-ruby/).

File: lib/integrity/branch.rb [hunk 2]
```diff
@@ -30,11 +31,13 @@ def set_story
...
+      logger.info "Error setting tracker story to branch error: #{e.message}"
```
Comment (suwyn) [inline lib/integrity/branch.rb]: We log errors with the error log level.   `logger.error`

**Comment by suwyn** [comment]
@kquet Is there ever any reason not to run a build if the tracker integration fails?

**Pull request closed without merging**